### PR TITLE
[Feature] Game Loop

### PR DESCRIPTION
**Pull request by mikelfcosta**
A lot was done.
We now have a game with passing only.

### CODE REVIEW

File: src/api/db/gameSession.ts [hunk 1]
```diff
@@ -52,3 +52,27 @@ export async function requestPlayerHandListener(
...
+export async function requestProgressionListener(
+  sessionId: string,
+  callback: (plays: Normalized<Play>) => void,
+) {
+  try {
+    await getSessionRef(sessionId)
+      .collection('progression')
+      .onSnapshot(querySnapshot => {
+        const progression = normalizeQuery<Play>(querySnapshot);
+        console.log({ progression });
```
Comment (davidpn11) [inline src/api/db/gameSession.ts:65]: log

File: src/components/ActionArea/styles.ts [hunk 2]
```diff
@@ -23,7 +23,6 @@ export const SectionWrapper = styled.div`
   height: 190px;
   padding: 8px;
   width: 150px;
-  cursor: pointer;
```
Comment (davidpn11) [inline src/components/ActionArea/styles.ts:26]: any reason to remove it?

File: src/components/Side/index.tsx [hunk 3]
```diff
@@ -1,45 +1,78 @@
...
-      const itemSlots = Math.floor(spaceToUse / HISTORY_ITEM_HEIGHT);
+      const othersHeight = heightRef.current.clientHeight;
+      const spaceToUse = wrapperHeight - othersHeight - 32;
+      const calculatedSlots = Math.floor(spaceToUse / (HISTORY_ITEM_HEIGHT + 8));
 
-      if (plays.length > itemSlots) setShowCount(true);
-      else setShowCount(false);
+      setItemSlots(calculatedSlots);
     }
-  }, [wrapperRef, historyRef, plays]);
```
Comment (davidpn11) [inline src/components/Side/index.tsx:27]: I believe if you add `historyRef.current.clientHeight` to the listener array would fix the resizing window bug

File: src/components/Side/index.tsx [hunk 4]
```diff
@@ -1,45 +1,78 @@
...
+              player={player}
+            />
+          ))}
+        </PlayersWrapper>
+        <Title>History</Title>
+      </HeightWrapper>
+      <HistoryWrapper>
+        {pipe(
+          plays,
+          A.reverse,
```
Comment (davidpn11) [inline src/components/Side/index.tsx:71]: Nice

File: src/views/Game/helpers/plays.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  if (O.isNone(currentPlayer)) throw new Error('Cannot get position of not playing player.');
+
+  const currentPlayerPosition = currentPlayer.value.position;
+
+  const lastPlayerPosition = Object.values(session.players).length - 1;
+  const currentDirection = session.direction;
+
+  let nextPlayerPosition = 0;
+  if (currentDirection === 'LEFT') {
+    nextPlayerPosition = isFirstPosition(currentPlayerPosition)
```
Comment (davidpn11) [inline src/views/Game/helpers/plays.ts:31]: Is this correct?
If I got this right, if the user is first position moves to the next.
If not, moves to the last position.

File: src/views/Game/hooks/useProgressionListener.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  const orderedProgression = useSelector(getOrderedProgression);
+  const [hasListener, setHasListener] = useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  function runNextEffect(play: PlayWithId) {
+    const nextPlayer = getNextPlayer(play, currentSession);
+
+    batch(() => {
+      dispatch(setCurrentPlay(play.id));
+      dispatch(setCurrentPlayer(nextPlayer.id));
```
Comment (davidpn11) [inline src/views/Game/hooks/useProgressionListener.tsx:38]: If we gonna have players making "out of their turn" plays, this might need some changes

Comment (davidpn11) [reply src/views/Game/hooks/useProgressionListener.tsx:38]: Or renaming for something more specific for a kind of play

File: src/views/Game/index.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  const { playerActions, handlePass } = useHandListener();
+  const { isCurrentPlayer } = useProgressionListener();
+
+  return (
+    <GameWrapper>
+      <Main>
+        <BoardWrapper>
+          <GameTable players={players} currentCard={currentCard} isActive={isCurrentPlayer} />
+        </BoardWrapper>
+        {/* These props will come from the Engine */}
```
Comment (davidpn11) [inline src/views/Game/index.tsx]: This comment can be removed

**Review by davidpn11 — APPROVED** [review]

File: src/components/ActionArea/styles.ts (hunk 2 above)
Comment (mikelfcosta) [reply src/components/ActionArea/styles.ts:26]: Yeah we don't really click on the currentCard, just on the Deck.

File: src/components/Side/index.tsx (hunk 3 above)
Comment (mikelfcosta) [reply src/components/Side/index.tsx:27]: historyRef.current might be null, so we cannot check for clientHeight

File: src/views/Game/helpers/plays.ts (hunk 5 above)
Comment (mikelfcosta) [reply src/views/Game/helpers/plays.ts:31]: Yes, on LEFT rotation it goes counter-clockwise so if the user is 0, the next one is the last position we have.

File: src/views/Game/hooks/useProgressionListener.tsx (hunk 6 above)
Comment (mikelfcosta) [reply src/views/Game/hooks/useProgressionListener.tsx:38]: Not sure I get it, when the player makes out of turn plays we dispatch the same things.

File: src/components/ActionArea/styles.ts (hunk 2 above)
Comment (davidpn11) [reply src/components/ActionArea/styles.ts:26]: :smart

**Pull request merged**